File: backend/app/schemas/project_library.py
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, Field


class ProjectLibraryCreate(BaseModel):
    name: str = Field(..., min_length=1, max_length=256)
    description: str | None = None


class ProjectLibraryUpdate(BaseModel):
    name: str | None = Field(None, min_length=1, max_length=256)
    description: str | None = None
    is_pinned: bool | None = None
    sort_order: int | None = None


class ProjectLibraryRead(BaseModel):
    id: int
    name: str
    description: str | None
    is_pinned: bool
    sort_order: int
    project_count: int = 0
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/project-library.ts
export type ProjectLibrary = {
  id: number
  name: string
  description: string | null
  is_pinned: boolean
  sort_order: number
  project_count: number
  created_at: string
  updated_at: string
}

export type ProjectLibraryCreate = {
  name: string
  description?: string | null
}

export type ProjectLibraryUpdate = {
  name?: string
  description?: string | null
  is_pinned?: boolean
  sort_order?: number
}
